Extension/src/LanguageServer/Providers/callHierarchyProvider.ts

```diff
@@ -125,73 +126,60 @@ export class CallHierarchyProvider implements vscode.CallHierarchyProvider {
 
     public async provideCallHierarchyIncomingCalls(item: vscode.CallHierarchyItem, token: vscode.CancellationToken):
         Promise<vscode.CallHierarchyIncomingCall[] | undefined | any> {
+        await this.client.awaitUntilLanguageClientReady();
+
+        // Cancel any current reference requests by firing a cancellation event to listeners.
+        workspaceReferences.cancelCurrentReferenceRequest(CancellationSender.NewRequest);
+
+        const CallHierarchyCallsToEvent: string = "CallHierarchyCallsTo";
+        if (item === undefined) {
+            this.logTelemetry(CallHierarchyCallsToEvent, CallHierarchyRequestStatus.Failed);
+            return undefined;
+        }
+
+        // Listen to VS Code cancellation.
+        let requestCanceled: CancellationSender = CancellationSender.None;
+        token.onCancellationRequested(e => { requestCanceled = CancellationSender.ProviderToken; });
+
+        // Process the request.
         return new Promise<vscode.CallHierarchyIncomingCall[] | undefined | any>((resolve, reject) => {
-            const CallHierarchyCallsToEvent: string = "CallHierarchyCallsTo";
-            if (item === undefined) {
-                this.logTelemetry(CallHierarchyCallsToEvent, CallHierarchyRequestStatus.Failed);
-                resolve(undefined);
+            // Listen to cancellation from an incoming new request, user or language server.
+            workspaceReferences.onCancellationRequested(sender => { requestCanceled = sender; });
+
+            // Define the callback that will process results.
+            const resultCallback: CallHierarchyResultCallback = (result: CallHierarchyCallsItemResult | null, progressBarDuration?: number) => {
+                if (requestCanceled !== CancellationSender.None || result === null || result?.calls === undefined) {

```

Comment:
This `result == null` check should not be required because `result?.call === undefined` will be true in that case.

Reply:
Fix pending. Fixed for line 160 too.